### ISSUE TITLE
Do you want live notifications when people reply to your posts? Enable Notifications​ ​ Liquibase 4.24.0 released! Pro Edition: Updates to Liquibase Flows, Structured Logging (Observability), and Security Improvements  Open Source and Pro Editions:  Database platform improvements for: SQL Anywhere, MySQL, MariaDB, Oracle, DB2 LUW, DB2 z/OS, PostgreSQL, H2, and HSQLDB. Plus: General Enhancements, Performance improvements, and JAVA API improvements see the Liquibase 4.24.0 Release Notes for additional details and download the latest release  Generated changelog has “add” foreign key constraint before “Drop” foreign Key constraint. liquibase jar version : 4.10

### ISSUE THREAD

**Issue by futtu** [liquibase/liquibase#5035]
### Search first

- [X] I searched and no similar issues were found

### Description

I have to upgrade liquibase-core jar from 3.4.2 to version > 4.8 for my application. In our application we are generating the changelog.xml via CLI command. we are using liquibase with Oracle PL/SQL

I tried with liquibase-core 4.10 version. But the generated diff changelog has wrong order of foreign key constraints.

Output from liquibase-core 3.4.2 version:

These are the only change log which got generated for this table for this version.

<changeSet author="oracle (generated)" id="1679476433218-565">
        <dropForeignKeyConstraint baseTableName="ENDPOINTPROBE_SUBNETS" constraintName="FKENDPNTPRB_SBNTS_SUBNET_ID"/>
        <addForeignKeyConstraint baseColumnNames="SUBNET_ID" baseTableName="ENDPOINTPROBE_SUBNETS" constraintName="FKENDPNTPRB_SBNTS_SUBNET_ID" referencedColumnNames="ID" referencedTableName="ENDPOINTPROBESUBNETS"/>
    </changeSet>
Output from liquibase-core 4.10.0 version:
This version has generated more number of change log. It has generated the change log for all the keys present in table. Point to note is that for all the other constraints drop has been generated first and then add. Only for foreign key it is other way round.

<changeSet author="oracle (generated)" id="1678988969531-465">
        <dropIndex indexName="FIENDPNTPRB_SBNTS_SUBNET_ID" tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-466">
        <createIndex indexName="FIENDPNTPRB_SBNTS_SUBNET_ID" tableName="ENDPOINTPROBE_SUBNETS" unique="false">
            <column name="SUBNET_ID"/>
        </createIndex>
    </changeSet>
     <changeSet author="oracle (generated)" id="1678988969531-489">
        <dropIndex indexName="FINDPNTPRB_SBNTS_PSNDPNTPRB" tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-490">
        <createIndex indexName="FINDPNTPRB_SBNTS_PSNDPNTPRB" tableName="ENDPOINTPROBE_SUBNETS" unique="false">
            <column name="ENDPOINTPROBE"/>
        </createIndex>
    </changeSet>
     <changeSet author="oracle (generated)" id="1678988969531-2041">
        <dropPrimaryKey tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-2042">
        <createIndex indexName="PKENDPOINTPROBE_SUBNETS" tableName="ENDPOINTPROBE_SUBNETS" unique="true">
            <column name="ENDPOINTPROBE"/>
            <column name="SUBNET_ID"/>
        </createIndex>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-2043">
        <addPrimaryKey columnNames="ENDPOINTPROBE, SUBNET_ID" constraintName="PKENDPOINTPROBE_SUBNETS" forIndexName="PKENDPOINTPROBE_SUBNETS" tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-3437">
        <addForeignKeyConstraint baseColumnNames="SUBNET_ID" baseTableName="ENDPOINTPROBE_SUBNETS" constraintName="FKENDPNTPRB_SBNTS_SUBNET_ID" deferrable="false" initiallyDeferred="false" onDelete="RESTRICT" onUpdate="RESTRICT" referencedColumnNames="ID" referencedTableName="ENDPOINTPROBESUBNETS" validate="true"/>
    </changeSet>
     <changeSet author="oracle (generated)" id="1678988969531-3451">
        <dropForeignKeyConstraint baseTableName="ENDPOINTPROBE_SUBNETS" constraintName="FKENDPNTPRB_SBNTS_SUBNET_ID"/>
    </changeSet>
While applying these change logs it errors out saying :

Caused by: Error : 2264, Position : 57, Sql = ALTER TABLE ENDPOINTPROBE_SUBNETS ADD CONSTRAINT FKENDPNTPRB_SBNTS_SUBNET_ID FOREIGN KEY (SUBNET_ID) REFERENCES ENDPOINTPROBESUBNETS (ID), OriginalSql = ALTER TABLE ENDPOINTPROBE_SUBNETS ADD CONSTRAINT FKENDPNTPRB_SBNTS_SUBNET_ID FOREIGN KEY (SUBNET_ID) REFERENCES ENDPOINTPROBESUBNETS (ID), Error Msg = ORA-02264: name already used by an existing constraint
The ddlsql for the table is

prompt create table EndPointProbe_subnets
    create table EndPointProbe_subnets(
        EndPointProbe varchar2(36) CONSTRAINT NN45447695037066403 NOT NULL,
        SUBNET_ID varchar2(36) CONSTRAINT NN45447695037083963 NOT NULL,
        CONSTRAINT PKEndPointProbe_subnets PRIMARY KEY (EndPointProbe, SUBNET_ID)
    );
    ALTER TABLE EndPointProbe_subnets ADD CONSTRAINT FKndPntPrb_sbnts_PSndPntPrb FOREIGN KEY (EndPointProbe) REFERENCES EndPointProbe (ID);
    ALTER TABLE EndPointProbe_subnets ADD CONSTRAINT FKEndPntPrb_sbnts_SUBNET_ID FOREIGN KEY (SUBNET_ID) REFERENCES EndPointProbeSubnets (ID);                              
    CREATE INDEX FIndPntPrb_sbnts_PSndPntPrb ON EndPointProbe_subnets (EndPointProbe);
    CREATE INDEX FIEndPntPrb_sbnts_SUBNET_ID ON EndPointProbe_subnets (SUBNET_ID);
        
Can someone help me in resolving this issue.

I tried with the different versions of liquibase - 4.0, 4.8 and 4.10. But all are generating the same changelog.xml

### Steps To Reproduce

While upgrading the liquibase-core version from 3.4.2 to 4.10.0. The order of constraints are not getting generated properly.

Output from liquibase-core 3.4.2 version:
These are the only change log which got generated for this table for this version.

<changeSet author="oracle (generated)" id="1679476433218-565">
        <dropForeignKeyConstraint baseTableName="ENDPOINTPROBE_SUBNETS" constraintName="FKENDPNTPRB_SBNTS_SUBNET_ID"/>
        <addForeignKeyConstraint baseColumnNames="SUBNET_ID" baseTableName="ENDPOINTPROBE_SUBNETS" constraintName="FKENDPNTPRB_SBNTS_SUBNET_ID" referencedColumnNames="ID" referencedTableName="ENDPOINTPROBESUBNETS"/>
    </changeSet>

Output from liquibase-core 4.10.0 version:
This version has generated more number of change log. It has generated the change log for all the keys present in table. Point to note is that for all the other constraints drop has been generated first and then add. Only for foreign key it is other way round.

<changeSet author="oracle (generated)" id="1678988969531-465">
        <dropIndex indexName="FIENDPNTPRB_SBNTS_SUBNET_ID" tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-466">
        <createIndex indexName="FIENDPNTPRB_SBNTS_SUBNET_ID" tableName="ENDPOINTPROBE_SUBNETS" unique="false">
            <column name="SUBNET_ID"/>
        </createIndex>
    </changeSet>
     <changeSet author="oracle (generated)" id="1678988969531-489">
        <dropIndex indexName="FINDPNTPRB_SBNTS_PSNDPNTPRB" tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-490">
        <createIndex indexName="FINDPNTPRB_SBNTS_PSNDPNTPRB" tableName="ENDPOINTPROBE_SUBNETS" unique="false">
            <column name="ENDPOINTPROBE"/>
        </createIndex>
    </changeSet>
     <changeSet author="oracle (generated)" id="1678988969531-2041">
        <dropPrimaryKey tableName="ENDPOINTPROBE_SUBNETS"/>
    </changeSet>
    <changeSet author="oracle (generated)" id="1678988969531-2042">
        <createIndex indexName="PKENDPOINTPROBE_SUBNETS" tableName="ENDPOINTPROBE_SUBNETS" unique="true">
            <column name="ENDPOINTPROBE"/>
            <column name="SUBNET_ID"/>
        </createIndex>

The ddlsql for the table is

prompt create table EndPointProbe_subnets
    create table EndPointProbe_subnets(
        EndPointProbe varchar2(36) CONSTRAINT NN45447695037066403 NOT NULL,
        SUBNET_ID varchar2(36) CONSTRAINT NN45447695037083963 NOT NULL,
        CONSTRAINT PKEndPointProbe_subnets PRIMARY KEY (EndPointProbe, SUBNET_ID)
    );
    ALTER TABLE EndPointProbe_subnets ADD CONSTRAINT FKndPntPrb_sbnts_PSndPntPrb FOREIGN KEY (EndPointProbe) REFERENCES EndPointProbe (ID);
    ALTER TABLE EndPointProbe_subnets ADD CONSTRAINT FKEndPntPrb_sbnts_SUBNET_ID FOREIGN KEY (SUBNET_ID) REFERENCES EndPointProbeSubnets (ID);                              
    CREATE INDEX FIndPntPrb_sbnts_PSndPntPrb ON EndPointProbe_subnets (EndPointProbe);
    CREATE INDEX FIEndPntPrb_sbnts_SUBNET_ID ON EndPointProbe_subnets (SUBNET_ID);
        

### Expected/Desired Behavior

The drop foreign key constraint should get generated first and then add foreign key constraint should get generated

### Liquibase Version

4.0, 4.8, 4.10

### Database Vendor & Version

oracle 19c

### Liquibase Integration

CLI

### OS and/or Infrastructure Type/Provider

Redhat linux 8.x

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by tati-qalified** [comment]
Hello @futtu, thank you for letting us know of this issue.
I have a few questions so I can test this more effectively:
1. I noticed that you put "Output from ...", but there's no code after that. Did you mean to add that output?
2. Is it your intention to drop all constraints and then re-add them?
3. In the case that you want to alter the constraint, would it work if you named them something else, other than the existing constraints?

Thank you
Tatiana

**Comment by futtu** [comment]
Hi @tati-qalified  
Thanks for looking into this issue. Sorry for bad formatting of the description. I have updated the format so that the description proper.
**1. I noticed that you put "Output from ...", but there's no code after that. Did you mean to add that output?**

Yes, the older version of liquibase(3.4.2) has generated only 2 changelogs as shown in the above. Also the order of changelog is fine. Whereas the newer version generates lots of changelogs and the order is not proper and hence creating exception

** 2. Is it your intention to drop all constraints and then re-add them?**
No, but the changelogs are getting generated that way only.


**In the case that you want to alter the constraint, would it work if you named them something else, other than the existing constraints?**
We are using the liquibase for generating the schema difference between the two versions of our application. 
Basically we are using liquibase for our upgrade process. It generates the changelog based on the schema difference and then we use that changelog to upgrade our DB + data.

**Comment by filipelautert** [comment]
This was tested on both Liquibase 4.10.0 and 4.24.0 and could not be reproduced — the foreign key constraint ordering worked as expected. Since there was no follow-up after the request to retest on a newer version, closing this out. If you're still seeing this issue on a current version, please open a new issue with updated details. Thanks for the report\!
